## 1. Clean your Datasets

For the first part of your project, you will apply the same data cleaning steps to your "data/control.txt" and "data/treatment.txt" files. By the end of this step, you should have two lists of floats that match the numbers you have in each respective file.

You will implement your `clean_data` function in your `cleaner.py` module and import the module in the code-block below.

In [8]:
# TODO: import all the packages that you will need for this notebook here
import cleaner as clean
import statistics as stats
import matplotlib.pyplot as plt
import numpy as np

ModuleNotFoundError: No module named 'matplotlib'

In [4]:
# TODO: read your text files into your code using file I/O

control_file = open('data/control.txt')
treatment_file = open('data/treatment.txt')
...

# TODO: clean both list objects that you created above using the 'clean_data' function
...
control = clean.clean_data(control_file)
treatment = clean.clean_data(treatment_file)

## 2. Explore your Datasets

Now that you have your list objects created and cleaned, let's explore the descriptive statistics of these two files and make subsequent observations on our data.

In [5]:
# TODO: Calculate the standard deviation of both the control and treatment group
...
control_stdev = stats.stdev(control)
treatment_stdev = stats.stdev(treatment)
# TODO: print your findings
...
print("Control Standard Deviation: ", round(control_stdev, 2))
print("Treatment Standard Deviation: ", round(treatment_stdev, 2))

Control Standard Deviation:  3.26
Treatment Standard Deviation:  2.09


In [6]:
# TODO: Calculate the mean of both the control and treatment group
...
control_mean = stats.mean(control)
treatment_mean = stats.mean(treatment)

# TODO: print your findings
...
print("Control Mean: ", round(control_mean, 2))
print("Treatment Mean: ", round(treatment_mean, 2))

Control Mean:  12.45
Treatment Mean:  16.8


In [7]:
# TODO: Are there any outliers in your control group? (use 2 * stdev to find these outliers)
outliers_control = []
lower_bound = control_mean - (2 * control_stdev)
upper_bound = control_mean + (2 * control_stdev)
for line in control:
    if line < lower_bound or line > upper_bound:
        outliers_control.append(line)
...
print(outliers_control)

[5.92, 4.36, 4.21, 20.2, 3.69, 3.83, 21.21, 5.07, 4.87, 5.28, 5.75, 20.18, 20.95, 4.12, 19.21, 5.51, 4.01, 20.57, 20.72, 3.71]


In [8]:
# TODO: Are there any outliers in your treatment group?
outliers_treatment = []
...
lower_bound = treatment_mean - (2 * treatment_stdev)
upper_bound = treatment_mean + (2 * treatment_stdev)
for line in treatment:
    if line < lower_bound or line > upper_bound:
        outliers_treatment.append(line)
print(outliers_treatment)

[21.17, 10.34, 21.07, 11.54, 21.66, 22.11, 12.02, 12.41, 11.35, 22.85, 21.49, 11.99, 21.57, 21.76, 12.54, 12.53, 12.34, 12.32, 21.42, 10.33]


## 3. Create Data Visualizations

For this final step, generate data visualizations to view the dispersion of both of your groups.

Make sure that the two visualizations you create are on the same figure. 

In [6]:
# TODO: create a data visualization to view differences in dispersion and mean

plt.figure(figsize=(10, 5))
plt.hist(control,  bins=10, range=(0, 25), color="orange", alpha = 0.6)
plt.hist(treatment, bins=10, range=(0, 25), color="green", alpha = 0.6)
plt.title("Histogram Showing Control and Treatment")
plt.legend()
plt.show()
...

NameError: name 'plt' is not defined

## 4. Your Data Findings

What did you discover about your two datasets? Which group has the highest viewing time? What would you want to investigate next?


The treatment has the highest viewing time. The control mean differed from the treatment mean as follows: The Control Mean was 12.45 and the Treatment Mean was 16.8. The Control Standard Deviation was 3.26 and the Treatment Standard Deviation was 2.09. The Control dataset is more spread out compared to the Treatment dataset.The Viewing times for the Control took less time compared to the treatment which means trainees would be more engaged if they could read the slides faster.